src/core/parse.js

```diff
@@ -280,17 +272,47 @@ foam.CLASS({
       var str = this.s;
       var ps = this.ps;
       for ( var i = 0 ; i < str.length ; i++, ps = ps.tail ) {
-        if ( str.charAt(i) !== ps.head ) {
+        if ( ! ps.head || str.charAt(i) !== ps.head.toLowerCase() ) {
           this.fail.ps = this.ps;
           return this.fail;
         }
       }
-      this.success.ps = ps.setValue(this.value || str);
+      this.success.ps = ps.setValue(this.value || this.s);
       return this.success;
     }
   ]
 });
 
+foam.CLASS({
+  package: 'foam.parse.compiled',
+  name: 'LiteralICWithValue',
+  extends: 'foam.parse.compiled.State',
+
+  properties: [
+    {
+      name: 's',
+      final: true
+    },
+    {
+      name: 'value',
+      final: true
+    }
+  ],
+  methods: [
+    function step() {
+      var str = this.s.toLowerCase();
```

Comment:
Shouldn't we do this on the postSet() of 's' so that we only have to do it once?